src/main/java/duke/manager/TaskManager.java

```diff
@@ -0,0 +1,197 @@
+            message = message.substring(9);
+            String[] deadlineData = message.split("/", 2);
+            tasks[Task.COUNT] = new Deadline(deadlineData[0], deadlineData[1].substring(3));
+            Task.COUNT++;
+        } catch (StringIndexOutOfBoundsException e) {
+            printLine();
+            System.out.println("YOU IDIOT !!??!! The description of a deadline cannot be empty.");
+            printLine();
+        } catch (ArrayIndexOutOfBoundsException d) {
+            printLine();
+            System.out.println("YOU IDIOT !!??!! The input format should be : ");
+            System.out.println("deadline description /by date or time");
+            printLine();
+        }
+    }
+
+    /**
+     * The function adds the event input by the user
+     *
+     * @param tasks   the array of tasks
+     * @param message the input string containing
+     *                the task description
+     */
+    private static void addTodo(Task[] tasks, String message) {
+        try {
+            tasks[Task.COUNT] = new Todo(message.substring(5));
+            Task.COUNT++;
+        } catch (StringIndexOutOfBoundsException e) {
+            printLine();
+            System.out.println("OOPS!!! The description of a todo cannot be empty.");
+            printLine();
+        }
+    }
+
+    /**
+     * Prints all the tasks
+     */
+    private static void printTasks(Task[] tasks) {
+        printLine();
+        for (int i = 0; i < Task.COUNT; i++) {
+            System.out.println((i + 1) + ". " + tasks[i].getDescription());
+        }
+        if (Task.COUNT == 0) {
+            System.out.println("Smartass, you need to add tasks before listing them !!!");
+        }
+        printLine();
+    }
+
+    /**
+     * Marks the task done based
+     * on the integer value provided
+     * by the user in the String message.
+     *
+     * @param tasks   the array of tasks
+     * @param message the input string containing
+     *                the task description
+     */
+    private static void markDone(Task[] tasks, String message) {
+        try {
+            String[] arrOfStr = message.split(" ");
+            int index = Integer.parseInt(arrOfStr[arrOfStr.length - 1]) - 1;
+            tasks[index].isDone();
+        } catch (NullPointerException e) {
+            printLine();
+            System.out.println("OH MY GOD, can you maybe type a task that exists ?");
+            printLine();
+        }
+    }
+
+    /**
+     * The function processes the inputs
+     * by the user and calls the necessary
+     * functions to add, print and manipulate
+     * tasks.
+     */
+    public static void processInput() {
+
+        Task[] tasks = new Task[Task.MAX];
+        Scanner scanner = new Scanner(System.in);
+        String message = scanner.nextLine();
+
+        while (!message.equals("bye")) {
+            if (message.equals("list")) {
+                printTasks(tasks);
+            } else if (message.contains("done")) {
```

Comment:
Using message.contains("done") to spot the done command seems a bit dangerous. What if the added Todo is "Get the job done" (which contains "done")?